internal/xds/rbac/rbac_engine.go

```diff
@@ -238,3 +274,30 @@ type rpcData struct {
 	// handshake.
 	certs []*x509.Certificate
 }
+
+func doAuditLogging(engine *engine, rpcData *rpcData, rule string, authorized bool) error {
+	event := &audit.Event{
+		FullMethodName: rpcData.fullMethod,
+		Principal:      rpcData.peerInfo.Addr.String(),
+		PolicyName:     engine.policyName,
+		MatchedRule:    rule,
+		Authorized:     authorized,
+	}
+	for _, logger := range engine.auditLoggers {
+		switch engine.auditCondition {
+		case v3rbacpb.RBAC_AuditLoggingOptions_NONE:
+			continue
+		case v3rbacpb.RBAC_AuditLoggingOptions_ON_DENY:
+			if !authorized {
+				(*logger).Log(event)
```

Comment:
Is this dereferencing needed? In Go, method invocations should work with `.` for either one.

Reply:
Yes, the compiler is unhappy otherwise
```
logger.Log undefined (type *audit.Logger is pointer to interface, not interface)compiler[MissingFieldOrMethod](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#MissingFieldOrMethod)
```

Reply:
Ah I see. So I think because the Logger is an interface itself, you don't need to store the pointers of it in the engine. The field should just be []audit.Logger. If it's the pointer of your impl that implements this interface, this slice will store the pointers anyway. And then you won't need this dereferencing.

Reply:
Ah I see, changed - PTAL